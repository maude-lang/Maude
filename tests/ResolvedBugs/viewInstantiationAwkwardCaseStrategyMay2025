#!/bin/sh

MAUDE_LIB=$srcdir/../../src/Main
export MAUDE_LIB

../../src/Main/maude -no-prelude \
  < $srcdir/viewInstantiationAwkwardCaseStrategyMay2025.maude -no-banner -no-advise \
  > viewInstantiationAwkwardCaseStrategyMay2025.out 2>&1

diff $srcdir/viewInstantiationAwkwardCaseStrategyMay2025.expected viewInstantiationAwkwardCaseStrategyMay2025.out > /dev/null 2>&1
