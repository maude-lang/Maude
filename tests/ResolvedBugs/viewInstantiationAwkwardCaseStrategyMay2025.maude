***
*** Counterpart of the viewInstantiationAwkwardCaseApril2025 test for
*** strategy to strategy expression mappings.
***

smod INT is
  sorts Nat Int .
  subsort Nat < Int .
  strat s : Int @ Int .
endsm

sth A is	
  sorts Small Big .
  subsort Small < Big .
  strat f : Big @ Big .
endsth

view INT-as-an-A from A to INT is
  sort Small to Nat .
  sort Big to Int .
  strat f(N:Big) to expr s(N:Int) .
endv

sth B is
  sort Card .
  strat g : Card @ Card .
endsth

smod BASE{X :: A} is
endsm

view BASE-as-a-B{X :: A} from B to BASE{X} is
  sort Card to X$Small .
  strat g to f .
endv

smod USE-B{X :: B} is
  sd g(N:X$Card) := match N:X$Card [label "raw thumb"] .
endsm

smod TEST is
  pr USE-B{BASE-as-a-B{INT-as-an-A}} .
endsm

show sds .
