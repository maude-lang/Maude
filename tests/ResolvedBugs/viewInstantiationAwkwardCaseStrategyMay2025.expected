sd s(N:Nat) := match N:Nat .
Bye.
